### PR TITLE
Add `strict` option

### PR DESCRIPTION
**Pull request by franciscocardoso**
This commit also contains code style updates, documentation improvements
and a rename to itin-validator.

Refs https://github.com/seegno/is-valid-itin/issues/2.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "presets": ["es2015"]
+  "presets": ["es2015-node4", "stage-1"]
```
Comment (ruimarinho) [inline .babelrc:2]: Not sure if stage-1 is required here.. can you confirm if we're using it?

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+CHANGELOG.md
```
Comment (ruimarinho) [inline .dockerignore:1]: File can be removed.

File: .jscsrc [hunk 3]
```diff
@@ -1,4 +1 @@
-plugins:
-  - 'jscs-config-seegno'
-
 preset: seegno
```
Comment (ruimarinho) [inline .jscsrc:1]: Rename to .yml.

File: README.md [hunk 4]
```diff
@@ -1,44 +1,116 @@
...
+This method validates if the given value is a valid `Individual Taxpayer Identification Number`.
 
+#### Arguments
+
+1. `value` *(&#42;)*: The value to validate.
+2. `[options]` *(Object)*: The options object.
+3. `[options.strict=true]` _(boolean|string)_: Whether or not formatting characters such as dashes or spaces should be rejected and if they must be in their precise location.
+
+#### Returns
+*(boolean)*:  Returns `true` if `value` is a valid Individual Taxpayer Identification Number, else `false`.
```
Comment (ruimarinho) [inline README.md:27]: Double space after :.

Comment (ruimarinho) [inline README.md:27]: Returns whether `value` is a valid Individual Taxpayer Identification Number?

File: dist/index.js [hunk 5]
```diff
@@ -3,47 +3,57 @@
 Object.defineProperty(exports, "__esModule", {
```
Comment (ruimarinho) [inline dist/index.js:3]: Dist should not be part of the PR.

File: package.json [hunk 6]
```diff
@@ -1,40 +1,59 @@
...
-  "name": "is-valid-itin",
-  "version": "0.0.2",
-  "description": "Validate Individual Taxpayer Identification Number",
-  "main": "dist/index.js",
-  "homepage": "https://github.com/seegno/is-valid-itin",
-  "bugs": "https://github.com/seegno/is-valid-itin/issues",
-  "repository": "seegno/is-valid-itin",
-  "author": "Seegno",
-  "license": "MIT",
+  "name": "itin-validator",
```
Comment (ruimarinho) [inline package.json:2]: Use `sort-package-json` if you haven't done so to sort the package.

File: package.json [hunk 7]
```diff
@@ -1,40 +1,59 @@
...
+    "taxpayer",
+    "validator"
   ],
-  "options": {
-    "mocha": "--compilers js:babel-register --recursive --require should"
+  "homepage": "https://github.com/seegno/itin-validator",
+  "bugs": "https://github.com/seegno/itin-validator/issues",
+  "license": "MIT",
+  "author": "Seegno",
+  "main": "dist/index.js",
```
Comment (ruimarinho) [inline package.json:18]: src/dist even if we just transpire the src for now? Allows including test in the future.

File: package.json [hunk 8]
```diff
@@ -1,40 +1,59 @@
...
-    "changelog": "github_changelog_generator --no-issues",
+    "changelog": "github_changelog_generator --no-issues --header-label='# Changelog' --future-release=v$npm_config_future_release && sed -i '' -e :a -e '$d;N;2,4ba' -e 'P;D' CHANGELOG.md",
+    "cover": "babel-node node_modules/.bin/isparta cover --report html _mocha -- $npm_package_options_mocha",
     "lint": "eslint src test && jscs src test",
-    "prepublish": "npm run lint && npm test && npm run build",
-    "test": "NODE_ENV=test mocha $npm_package_options_mocha"
+    "prepublish": "npm run transpile",
+    "test": "mocha $npm_package_options_mocha",
+    "testdocker": "docker-compose run --rm sut",
+    "transpile": "rm -rf dist/* && babel src --out-dir dist",
```
Comment (ruimarinho) [inline package.json:30]: dist/src?

File: package.json [hunk 9]
```diff
@@ -1,40 +1,59 @@
...
+    "isparta": "^4.0.0",
+    "eslint-plugin-sort-class-members": "^1.0.1",
+    "jscs": "^2.11.0",
+    "jscs-config-seegno": "^2.0.0",
+    "mocha": "^2.3.4",
+    "pre-commit": "^1.1.2",
+    "should": "^8.0.0"
+  },
+  "engines": {
+    "node": ">=0.10"
```
Comment (ruimarinho) [inline package.json:51]: > 4?

File: test/index_test.js [hunk 10]
```diff
@@ -3,88 +3,97 @@
...
-  '900949999',
-  '999949999',
-  '900990000',
-  '900999999',
-  '999999999'
-];
+const numbers = {
+  invalid: ['00070000', '100_70_000', '20070000f', '300#70#000'],
+  valid: {
+    format: ['900-70-0000', '900 70 9999', '999-70 9999', '900 88 0000'],
```
Comment (ruimarinho) [inline test/index_test.js:16]: We've discussed this before but after looking at moment's isValid/strict option, I think we should consider making `strict` validate the numbers + format and delimiters instead. TBD.

File: README.md [hunk 11]
```diff
@@ -1,44 +1,116 @@
...
+```
+
+## License
+MIT
+
+## Credits
+
+Many thanks to [miguelmota/is-valid-ssn](https://github.com/miguelmota/is-valid-ssn) for the original inspiration.
+
+[npm-image]: https://img.shields.io/npm/v/ssn-validator.svg?style=flat-square
```
Comment (ruimarinho) [inline README.md:113]: ssn-validator -> is-valid-tin

File: README.md [hunk 12]
```diff
@@ -1,44 +1,116 @@
...
+### `mask(value, [options])`
 
-> Mask the number.
+This method will help you protect this sensitive piece of information by obfuscating some digits.
 
+#### Arguments
+
+1. `value` *(&#42;)*: The value to mask.
+2. `[options]` *(Object)*: The options object.
+3. `[options.strict=true]` _(boolean|string)_: Whether or not formatting characters such as dashes or spaces should be rejected and if they must be in their precise location.
```
Comment (ruimarinho) [inline README.md:59]: I would just mention that this is forwarded to isValid.

**Comment by ruimarinho** [comment]
We will be deprecating this PR in favour of removing `options`. This will allow us keeping the module simpler and considering `value` as valid if it includes numbers only (in the expected range) or a formatted string as `XXX-XX-XXXX`.

**Comment by franciscocardoso** [comment]
Deprecated by https://github.com/seegno/is-valid-itin/pull/5.
